File: codewars/kyu_6/python/persistent_bugger.py
# https://www.codewars.com/kata/persistent-bugger/
import functools
import operator


def persistence(n):
    counter = 0
    while n > 10:
        n = int(functools.reduce(operator.mul, [int(x) for x in str(n)], 1))
        counter += 1
    return counter

File: codewars/kyu_6/php/persistent_bugger.php
<?php

# https://www.codewars.com/kata/persistent-bugger/

function persistence(int $num): int {
  $count = 0;
  while($num > 9) {
    $num = array_product(str_split($num));
    $count++;
  }
  return $count;
}

?>
